internal/character/gepard/talent.go

```diff
@@ -0,0 +1,74 @@
+package gepard
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	Talent key.Modifier = "gepard-talent"
+)
+
+type talentState struct {
+	revivePerc float64
+	a4Active   bool
+	e6Active   bool
+}
+
+func init() {
+	modifier.Register(Talent, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnLimboWaitHeal: func(mod *modifier.ModifierInstance) bool {
+
+				// Dispel all debuffs
+				mod.Engine().DispelStatus(mod.Owner(), info.Dispel{
+					Status: model.StatusType_STATUS_DEBUFF,
+					Order:  model.DispelOrder_LAST_ADDED,
+				})
+
+				// Queue Heal
+				mod.Engine().InsertAbility(info.Insert{
+					Execute: func() {
+						mod.Engine().SetHP(
+							mod.Owner(), mod.Owner(), mod.OwnerStats().MaxHP()*mod.State().(talentState).revivePerc)
+					},
+					Source:   mod.Owner(),
+					Priority: info.CharReviveSelf,
+				})
+
+				// If A4, restore Energy to 100% (Energy Cost is 100)
+				if mod.State().(talentState).a4Active {
+					mod.Engine().ModifyEnergyFixed(mod.Owner(), 100)
+				}
+
+				// If E6, action forward
+				if mod.State().(talentState).e6Active {
+					mod.Engine().SetGauge(mod.Owner(), 0)
+				}
+
+				mod.RemoveSelf()
```

Comment:
these should be part of the `InsertAbility` Execute func (only happen if the revive goes through)

nit: fine with leaving the E6 as-is, but may need some changes. Need to follow up with some TC since it seems to have a weird edge case based on DM